app/src/main/java/ua/dancecamp/vmedvediev/ua21summerdancecamp/UI/splash/SplashActivity.kt

```diff
@@ -0,0 +1,123 @@
+                if (it != null) {
+                    val localeLanguage = baseContext.resources.configuration.locale.language
+                    val applicationLanguage = if (it.interfaceLanguage.isEmpty()) {
+                        localeLanguage
+                    } else {
+                        it.interfaceLanguage
+                    }
+                    val previousLocaleLanguage = if (it.localeLanguage.isEmpty()) {
+                        localeLanguage
+                    } else {
+                        it.localeLanguage
+                    }
+                    when {
+                        EventsCache.eventsList.isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                            finish()
+                        }
+                        splashViewModel.getEvents().isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                            splashViewModel.setupLocalStorage()
+                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                            finish()
+                        }
+                        else -> JsonParserAsyncTask(WeakReference(this@SplashActivity)).execute(applicationLanguage)
+                    }
+                    updateLocale(applicationLanguage)
+                    splashViewModel.saveApplicationSettigns(ApplicationSettings(it.id, applicationLanguage, applicationLanguage))
+                }
+            })
+        }
+    }
+
+    private fun updateLocale(applicationLanguage: String) {
+        val locale = Locale(applicationLanguage)
+        Locale.setDefault(locale)
+        val config = baseContext.resources.configuration
+        config.setLocale(locale)
+        baseContext.resources.updateConfiguration(config, baseContext.resources.displayMetrics)
+    }
+
+    private fun prepareEventsData(eventsList: ArrayList<Event>) {
+        splashViewModel.apply {
+            saveEvents(eventsList)
+            setupLocalStorage()
+        }
+    }
+
+    companion object {
+        class JsonParserAsyncTask(private val splashActivity: WeakReference<SplashActivity>) : AsyncTask<String, Void, ArrayList<Event>>() {
+
+            override fun doInBackground(vararg params: String?): ArrayList<Event>? {
+                Thread.sleep(1000)
+                 return try {
+                    val applicationInstance = MyApplication.instance
+                    val applicationContext = applicationInstance.applicationContext
+                    val applicationLanguage = params[0]
+                    val applicationAssets = applicationContext.assets
+
+                    val inputStream = when (applicationLanguage) {
+                        MyApplication.instance.getString(R.string.label_english) -> applicationAssets.open(applicationInstance.getString(R.string.list_of_events_en_json))
```

Comment:
Why do you use `MyApplication.instance` when you already have a separate variable for it?